### PR TITLE
Add federated tests in TypeScript using stop request in federated execution

### PR DESCRIPTION
**Pull request by hokeun**
Relevant issue: https://github.com/lf-lang/reactor-ts/issues/101

Relevant reactor-ts PR: https://github.com/lf-lang/reactor-ts/pull/111

### CODE REVIEW

**Comment by petervdonovan** [comment]
CI is failing due to #1311... just cross-referencing here in case anyone missed it.

**Review by petervdonovan — APPROVED** [review]
This looks mostly right to me. I just left minor comments.

File: test/TypeScript/src/Stop.lf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    timeout: 11 msec
+};
+
+import Sender from "lib/LoopedActionSender.lf"
+
+reactor Consumer {
+    input in1:number;
+    state reactionInvokedCorrectly:boolean(false);
+    reaction(in1) {=
+        let currentTag = util.getCurrentTag();
```
[the remark this answers is not in the document]
Comment (ByeongGil-Jun) [reply test/TypeScript/src/Stop.lf]: Thank you for your review and suggestion.
This comment was resolved in 461744a!

File: test/TypeScript/src/federated/DistributedStop.lf [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        // Sender should have requested stop earlier than the receiver.
+        // Therefore, the shutdown events must occur at (1 usec, 0) on the
+        // receiver.
+        if (!util.getElapsedLogicalTime().isEqualTo(TimeValue.usec(1)) ||
+            util.getCurrentTag().microstep !== 1) {
+            util.requestErrorStop(`ERROR: Receiver failed to stop the federation at the right time. `
+            + `Stopping at (${util.getElapsedLogicalTime()}, ${util.getCurrentTag().microstep}).`);
+        } else if (reaction_invoked_correctly === false) {
+            console.log("Receiver reaction(in) was not invoked the correct number of times. "
+            + `Stoping at (${util.getElapsedLogicalTime()}, ${util.getCurrentTag().microstep})`);
```
[the remark this answers is not in the document]
Comment (ByeongGil-Jun) [reply test/TypeScript/src/federated/DistributedStop.lf]: This is resolved in 5e6b535!

**Pull request merged**